pkg/products/observability/reconciler.go

```diff
@@ -0,0 +1,369 @@
+	r.log.Info("Start Observability reconcile")
+
+	operatorNamespace := r.Config.GetOperatorNamespace()
+	productNamespace := r.Config.GetNamespace()
+
+	phase, err := r.ReconcileFinalizer(ctx, client, installation, string(r.Config.GetProductName()), func() (integreatlyv1alpha1.StatusPhase, error) {
+		// Check if productNamespace is still present before trying to delete it resources
+		_, err := resources.GetNS(ctx, productNamespace, client)
+		if !k8serr.IsNotFound(err) {
+			// Mark OO CR for deletion.
+			phase, err := r.deleteObservabilityCR(ctx, client, installation, productNamespace)
+			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+				return phase, err
+			}
+
+			phase, err = resources.RemoveNamespace(ctx, installation, client, productNamespace, r.log)
+			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+				return phase, err
+			}
+		}
+		// Check if operatorNamespace is still present before trying to delete it resources
+		_, err = resources.GetNS(ctx, operatorNamespace, client)
+		if !k8serr.IsNotFound(err) {
+			phase, err := resources.RemoveNamespace(ctx, installation, client, operatorNamespace, r.log)
+			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+				return phase, err
+			}
+		}
+		// If both namespaces are deleted, return complete
+		_, operatorNSErr := resources.GetNS(ctx, operatorNamespace, client)
+		_, productNSErr := resources.GetNS(ctx, productNamespace, client)
+		if k8serr.IsNotFound(operatorNSErr) && k8serr.IsNotFound(productNSErr) {
+			return integreatlyv1alpha1.PhaseCompleted, nil
+		}
+		return integreatlyv1alpha1.PhaseInProgress, nil
+	}, r.log)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile finalizer", err)
+		return phase, err
+	}
+
+	phase, err = r.ReconcileNamespace(ctx, operatorNamespace, installation, client, r.log)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s ns", operatorNamespace), err)
+		return phase, err
+	}
+
+	phase, err = r.ReconcileNamespace(ctx, productNamespace, installation, client, r.log)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s ns", productNamespace), err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileConfigMap(ctx, client)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s configmap which is required to disable observability operator initilisting it's own cr", ConfigMapNoInit), err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileSubscription(ctx, client, installation, productNamespace, operatorNamespace)
```

Comment:
```suggestion
	phase, err = r.reconcileSubscription(ctx, client, productNamespace, operatorNamespace)
```

Since `installation` is part of the `Reconciler` struct, and `reconcileSubscription` is part of the struct methods, we should be able to remove the need to pass the `installation` as a required parameter and it can be accessed directly using `r.installation` if required in `reconcileSubscription` method :thinking:

Reply:
Fixed